python/hsfs/feature_view.py

```diff
@@ -78,6 +79,21 @@ def delete(self):
         """
         self._feature_view_engine.delete(self.name, self.version)
 
+    @staticmethod
+    def clean(feature_store_id, feature_view_name, feature_view_version):
+        """Delete the feature view and all associated metadata.
```

Comment:
Same comment as above, why not make this a function of FeatureStore?

Also, I ran into an issue when passing a `fs` instead of a `fs.id`. The error message was not to easy to interpret:

`RestAPIError: Metadata operation error: (url: https://hopsworks.glassfish.service.consul:8182/hopsworks-api/api/project/119/featurestores/%3Chsfs.feature_store.FeatureStore%20object%20at%200x7f02dc016c10%3E/featureview/cc_trans_fraud_all/version/12). Server response: 
HTTP code: 404, HTTP reason: Not Found, body: b'{"type":"restApiJsonResponse","errorCode":120004,"errorMsg":"Web application exception occurred","usrMsg":"HTTP 404 Not Found"}', error code: 120004, error msg: Web application exception occurred, user msg: HTTP 404 Not Found`

I wish we could do this. But I think this would cause a circular dependency between FeatureStore and FeatureView. 
```if isinstance(feature_store_id, FeatureStore):
      feature_store_id = feature_store_id.id

Maybe we can just add type hints and and/or check if feature_store_id is of type `int`.

Reply:
I would also call it "delete_feature_view" is this context. I think this would be more consistent with out the rest of the API. We usually use the term "delete..." , not "clean...". Cleans sounds like somnething is pruned from the featureview, but this action is to actually delete the feature view itself, right?

Reply:
The nuance is a little bit different. This methods "remove/delete" a **broken** fv. So maybe a different name can help users differentiate the two cases.

Reply:
Ok understood